src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -917,6 +917,62 @@ public EventAccessorInfo(MethodInfo interfaceAddMethod, MethodInfo interfaceRemo
                 {
                     InterfaceAddMethod = interfaceAddMethod;
                     InterfaceRemoveMethod = interfaceRemoveMethod;
+
+            private sealed class MethodInfoEqualityComparer : EqualityComparer<MethodInfo>
+            {
+                public static readonly MethodInfoEqualityComparer Instance = new MethodInfoEqualityComparer();
+
+                private MethodInfoEqualityComparer() { }
+
+                public override bool Equals(MethodInfo left, MethodInfo right)
+                {
+                    if (ReferenceEquals(left, right))
+                        return true;
+
+                    if (left == null)
```

Comment:
It's legal to have both of these defined in the same type. 

        public void Foo() {}
        public void Foo<T>() {}

So you may want to add a check for generic vs. non-generic methods. Since accessor methods will never be generic, it may be better to filter that out in your main loop rather than adding logic to the comparator to handle these. Your call.

Also, I wonder if it's worth checking the method attributes (such as "static" vs. "instance") C# seems to disallow this:

   public static void Foo() {}
   public void Foo() {}

but I'm not sure if this is a C# restriction or an ECMA restriction.